ui/client/containers/EspApp.js

```diff
@@ -24,17 +24,25 @@ import '../assets/fonts/fonts.less'
 import '../stylesheets/main.styl'
 import '../app.styl'
 import ErrorHandler from "./ErrorHandler"
+import {v4 as uuid4} from "uuid";
 
 class EspApp extends React.Component {
 
   componentDidMount() {
+    this.generateUniqueUserId()
     this.mountedHistory = this.props.history.listen((location, action) => {
       if (action === "PUSH") {
         this.props.actions.urlChange(location)
       }
     })
   }
 
+  generateUniqueUserId() {
+    if (!localStorage.getItem(userId)) {
```

Comment:
Look at SystemUtils and put evrything connected with localStorage in this class.

Reply:
Done